ttl/ttlworker/job_manager.go

```diff
@@ -498,25 +503,53 @@ tblLoop:
 		}
 
 		status := m.tableStatusCache.Tables[table.ID]
-		ok := m.couldTrySchedule(status, table, now, false)
-		if ok {
+		if m.couldLockJob(status, table, now, true, true) {
+			tables = append(tables, table)
+			isCreate = append(isCreate, true)
+		} else if m.couldLockJob(status, table, now, false, false) {
 			tables = append(tables, table)
+			isCreate = append(isCreate, false)
 		}
 	}
 
-	return tables
+	return tables, isCreate
 }
 
-// couldTrySchedule returns whether a table should be tried to run TTL
-func (m *JobManager) couldTrySchedule(tableStatus *cache.TableStatus, table *cache.PhysicalTable, now time.Time, ignoreScheduleInterval bool) bool {
-	if tableStatus == nil {
-		// if the table status hasn't been created, return true
-		return true
-	}
+// couldLockJob returns whether a table should be tried to run TTL
+func (m *JobManager) couldLockJob(tableStatus *cache.TableStatus, table *cache.PhysicalTable, now time.Time, isCreate bool, checkScheduleInterval bool) bool {
```

Comment:
How about splitting `couldLockJob` into something like `couldLockNewJob` and `couldLockExistingJob'?

Reply:
`couldLockNewJob` and `couldLockExistingJob` share quite a few codes, and they will have slightly different arguments.